### DISCUSSION TITLE
Another postgres compatibility exception to list?

### DISCUSSION THREAD

**Discussion by EvgSkv** [duckdb/duckdb#13451] [Q&A]
Hello,
I thought that compatibility of duckdb SQL with postgres SQL syntax is more like an aspiration than a strict goal for duckdb. But yesterday at DuckCon Hannes said that you actually test it.

So I am wondering if difference in `unnest` behaviour should be explicitly listed at [compatibility exceptions](https://duckdb.org/docs/sql/dialect/postgresql_compatibility.html) page?

Specifically when in Postgres I have an option to use `unnest` to run over array elements in the from clause like so:
```
select 10 * x from unnest(array[1,2,3]) as x;
```

However duckdb seems to understand this construct differently and when I run this I am getting an error:

```
duckdb> select 10 * x  from unnest(array[1,2,3]) as x;  
Binder Error: No function matches the given name and argument types '*(INTEGER_LITERAL, STRUCT("(ARRAY[1, 2, 3])" INTEGER))'. You might need to add explicit type casts.
```

I have a workaround, but I want to understand what is the reason for the error and how much is duckdb trying to be compatible with postgres syntax.

Thank you!

**Comment by pkoppstein** [comment]
Since in DuckDB we can write:
```
select 10 * x from (select unnest([1,2,3]) as x);
```
or even:
```
select 10 * unnest([1,2,3]) as x;
```
it would appear that the key point here is that in DuckDB, the `SELECT` in "... FROM SELECT..." cannot be dropped.
